pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -73,18 +75,79 @@ func (pl *CSILimits) Name() string {
 	return CSIName
 }
 
-// EventsToRegister returns the possible events that may make a Pod
+// EventsToRegister returns the possible events that may make a Pod.
 // failed by this plugin schedulable.
 func (pl *CSILimits) EventsToRegister() []framework.ClusterEventWithHint {
 	return []framework.ClusterEventWithHint{
 		// We don't register any `QueueingHintFn` intentionally
 		// because any new CSINode could make pods that were rejected by CSI volumes schedulable.
 		{Event: framework.ClusterEvent{Resource: framework.CSINode, ActionType: framework.Add}},
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}},
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}, QueueingHintFn: pl.isSchedulableAfterPodDeleted},
 		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add}},
+		// There is a possibility that a PVC is deleted when isSchedulableAfterPodDeleted is called.
+		// In this case, it's impossible if PVCs are relevant to the scheduling pod.
+		// To catch this case, we register PVC/Delete event.
```

Comment:
It's too vague to understand in which case `isSchedulableAfterPodDeleted` could miss the event; can you elaborate a bit more here? (like `valumeLimitKey` stuff)
I'd be easier to understand if you could include some examples here as well.

Reply:
How about this example?
https://github.com/kubernetes/kubernetes/pull/121508/commits/40a257d3e94466de2482c4a6e09daba1e44d46ee

Reply:
Actually I expected a more specific one, like:

```
		// If a deleted Pod has PVC, but PVC is deleted before Pod is deleted,
		// isSchedulableAfterPodDeleted cannot make a decision correctly.
		// e.g.,
		//  - An unscheduled Pod-A has PVC-A, and got rejected by CSILimits plugin.
		//  - Pod-B has PVC-B, and is running.
		//  - PVC-A and PVC-B get the same volumeLimitKey.
		// In this case, when Pod-B/PVC-B is deleted, if the scheduler receives Pod-B's deletion event after PVC-B is deleted,
		// it's impossible to determine whether Pod-A should be requeued or not in isSchedulableAfterPodDeleted,
		// which actually we should requeue in that case.
```

Reply:
Thanks, @sanposhiho. It would be cool, I'll copy & paste.

Reply:
I edited your suggestion a bit.
https://github.com/kubernetes/kubernetes/pull/121508/commits/baaaefdcb3a0eae5fbcf3874dad90775cc41801d